### DISCUSSION TITLE
.NET 7 slower than .NET 6 in some scenarios?

### DISCUSSION THREAD

**Discussion by kevinmcfarlane** [dotnet/runtime#78631] [Q&A] [answered]
I have a hand-rolled machine learning example adapted from a book. I timed this in .NET 4.8, 6 and 7. It roughly doubled in speed from 4.8 to 6, but slowed from 6 to 7.

4.8 1s
6.0 0.55s
7.0 0.77s

The Main function is
```
using DigitsRecognizerCore;

DateTime t0 = DateTime.Now;

var distance = new ManhattanDistance();
var classifier = new BasicClassifier(distance);

var trainingPath = @"..\..\..\Data\trainingsample.csv";
var training = DataReader.ReadObservations(trainingPath);
classifier.Train(training);

var validationPath = @"..\..\..\Data\validationsample.csv";
var validation = DataReader.ReadObservations(validationPath);

var correct = Evaluator.Correct(validation, classifier);
Console.WriteLine("Correctly classified: {0:P2}", correct);

Console.WriteLine("Time elapsed = {0:f} sec.", (DateTime.Now - t0).TotalSeconds);

Console.ReadLine();
```
The two csv files are about 9000KB and 55KB respectively.
I also tried cleaning and rebuilding and re-doing in a brand new .NET 7 project.
No change in results.

Any ideas?

**Comment by EgorBo** [comment, marked as answer]
@kevinmcfarlane can you run it from console with `DOTNET_TC_QuickJitForLoops=0` env var defined?
Another option to test is `DOTNET_GCName=clrgc.dll`

**Reply by kevinmcfarlane** [reply]
I tried the first option. That speeds it up slightly, although 6.0 is still a bit faster. I read somewhere that in 6.0, setting that to 0 speeds things up but that in .NET 7 there's no difference. Though to me it looks like it might be the other way round!

So for 6.0 I get 0.53/0.54s (two successive runs). For 7.0 it is 0.57/0.58s.

Still, nothing worth bothering about.

Thanks for your help. 👍
